app/src/main/java/com/vultisig/wallet/ui/navigation/NavController.kt

```
@@ -11,8 +11,19 @@ internal fun NavController.route(route: String, opts: NavigationOptions? = null)
     if (route == Destination.Back.route) {
         popBackStack()
     } else {
-        navigate(route) {
-            buildOptions(this, opts)
+        try {
+            navigate(route) {
+                buildOptions(
+                    this,
+                    opts
+                )
+            }
+        } catch (e: IllegalArgumentException) {
+            Timber.e(
+                e,
+                "Navigation failed for route: $route"
+            )
+            error("Navigation failed for route: $route")
```

will this throw the error again?

---

Yup, but this time I want to catch the exact route causing the exception.
I could set a default route to fall back on when an exception occurs, but I believe it's better to surface the actual details of the problem rather than masking it with fallback logic.

---

There might be an unknown error occurring on some devices that has been reported.